api/tinygo/runnable/db/query/query.go

```diff
@@ -0,0 +1,23 @@
+//go:build tinygo.wasm
+
+package query
+
+type Argument struct {
+	Name, Value string
+}
+
+func NewArgument(name, value string) Argument {
+	return Argument{
+		Name:  name,
+		Value: value,
+	}
+}
+
+type QueryType int32
+
+const (
+	QueryInsert QueryType = iota
```

Comment:
I would add a sentinel value here, so the "undefined zero value" doesn't default to `QueryInsert`.

See more at https://threedots.tech/post/safer-enums-in-go/#sentinel-values

Reply:
hm... given that we're still in beta, I think this would be one of the worthwhile "everything is broken, please recompile your things" changes here. Would save a lot of invisible bugs.

Reply:
I don't know if you remember when this exact thing happened with the HTTP method enum, but it was extremely painful. I'm not sure it's really worth it since the values are only being used internally. It's not really something a user could mess up (only us).

Reply:
Hm, fair. I think it would still be a good idea to break them out into an int / string backed set of constants at least.

```
# in errors.go
const (
    RunTimeError = 500
)

// elsewhere in the codebase
errors.RunTimeError // more easily understood

500 // what does it mean?
```

Reply:
For sure. We just need to settle on what those values are.